frontend/src/Components/ForgotPassword/ConfirmPassword.jsx

```diff
@@ -0,0 +1,108 @@
+   };
+
+   const onSuccessfulChange = async () => {
+      try {
+
+      } catch {
+         
+      }
+   };
+
+   const onFormSubmit = (e) => {
+      e.preventDefault();
+      if (password.match(RegEx) && password === confpassword) {
+         setSuccessMsg("Password changed successfully");
+         onSuccessfulChange();
+         redirectToLoginPage(1500);
+      } else {
+         setErrorMsg("Passwords do not match");
+      }
+   };
+
+   const passwordShowHide = () => {
+      setShowPassword(!showPassword);
+   };
+
+   const onSuccessfulValidation = () => {
+      if (password.match(RegEx) && password === confpassword) {
+         return true;
+      }
+      return false;
+   };
+
+   return (
+      <CenteredContainer>
+         <h4 className="text-center mb-5">Enter your new password</h4>
+         <p className={errorMsg ? "alert alert-danger text-danger text-center" : "d-none"}>{errorMsg}</p>
+         <p className={successMsg ? "alert alert-success text-success text-center" : "d-none"}>{successMsg}</p>
+         <form onSubmit={onFormSubmit}>
+            <div className="row">
+               <div className="col-md-6">
+                  <div className="form-outline mb-1">
+                     <Label>Password</Label>
+                     <div className="input-group">
+                        <input
+                           type={showPassword ? "text" : "password"}
+                           name="password"
+                           value={password}
+                           onChange={(e) => setPassword(e.target.value)}
+                           id="passwordInput"
+                           className="form-control form-control border-right-0"
+                           required
+                           autoComplete="off"
+                        />
+                        <div className="input-group-prepend">
+                           <span onClick={passwordShowHide} className="input-group-text bg-white rounded-right border-left-0">
+                              <i className={showPassword ? "fa-solid fa-eye" : "fa fa-eye"}></i>
+                           </span>
+                        </div>
+                     </div>
+                  </div>
+               </div>
+               <div className="col-md-6">
+                  <div className="form-outline mb-1">
+                     <Label>Confirm password</Label>
+                     <input
+                        type={showPassword ? "text" : "password"}
+                        name="confPassword"
+                        value={confpassword}
+                        onChange={(e) => setConfpassword(e.target.value)}
+                        id="confPasswordInput"
+                        className="form-control form-control"
+                        required
+                        autoComplete="off"
+                     />
+                  </div>
+               </div>
+            </div>
+            {onSuccessfulValidation() ? (
```

Comment:
There is no need for this to be defined as a function as it will be invoked at each render:
it can be defined as above:
const isPasswordValid = password.match(RegEx) && password === confpassword

and used here as { isPasswordValid ? (...) : (...) }